client/lobby/OptionsTab.cpp

```diff
@@ -43,20 +44,147 @@
 OptionsTab::OptionsTab() : humanPlayers(0)
 {
 	recActions = 0;
+		
+	addCallback("setTimerPreset", [&](int index){
+		if(!variables["timerPresets"].isNull())
+		{
+			auto tpreset = variables["timerPresets"].Vector().at(index).Vector();
+			TurnTimerInfo tinfo;
+			tinfo.baseTimer = tpreset.at(0).Integer() * 1000;
+			tinfo.turnTimer = tpreset.at(1).Integer() * 1000;
+			tinfo.battleTimer = tpreset.at(2).Integer() * 1000;
+			tinfo.creatureTimer = tpreset.at(3).Integer() * 1000;
+			CSH->setTurnTimerInfo(tinfo);
+		}
+	});
 	
-	addCallback("setTurnLength", std::bind(&IServerAPI::setTurnLength, CSH, _1));
+	auto parseTimerString = [](const std::string & str){
```

Comment:
Can you explain what this function does?
Why using `scanf`/`stoi` or boost lexical_cast was not an option?

Reply:
will add a comment